bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3298,7 +3298,6 @@ void ReplicaImp::onMessage<ReplicaRestartReadyMsg>(ReplicaRestartReadyMsg *msg)
   if (restart_ready_msgs_.size() == targetNumOfMsgs) {
     LOG_INFO(GL, "Target number = " << targetNumOfMsgs << " of restart ready msgs are recieved. Send resatrt proof");
     sendReplicasRestartReadyProof();
-    ControlStateManager::instance().onRestartProof(msg->seqNum());
```

Comment:
As we async sending the messages, I prefer to have the replica handle the restart messages only when it gets it from itself (this way we reduce the probability to restart before completing the actual sending)